### ISSUE TITLE
End-to-end regression test between metaseq and huggingface

### ISSUE THREAD

**Issue by shijie-wu** [facebookresearch/metaseq#136]
We are interested in a full regression test between metaseq and huggingface. As @patrickvonplaten confirmed, the current regression test doesn't cover the merging step. We observed different behaviors between models that require merging (1.3B) and the one that doesn’t (350M) (https://github.com/huggingface/transformers/issues/17545). It’s unclear what causes the difference at the moment due to the lack of an end to end regression test. 

To run the full regression test, we would need to load the model into metaseq directly. However, this is not well supported currently (e.g. https://github.com/facebookresearch/metaseq/issues/78 and https://github.com/facebookresearch/metaseq/issues/31). It would be great if people at Meta could help perform a full regression test between metaseq and huggingface 😄 

@suchenzang @stephenroller

**Comment by patrickvonplaten** [comment]
@shijie-wu could you specify a bit what is meant by 

> "We observed different behaviors between models that require merging and the ones that don't"

What exactly is different?

Thanks!

**Comment by shijie-wu** [comment]
> @shijie-wu could you specify a bit what is meant by 
> 
> 
> 
> > "We observed different behaviors between models that require merging and the ones that don't"
> 
> 
> 
> What exactly is different?
> 
> 
> 
> Thanks!

Hi @patrickvonplaten 

I'm referring to https://github.com/huggingface/transformers/issues/17545

**Comment by patrickvonplaten** [comment]
@shijie-wu, could you by any chance reproduce the result using the metaseq code base - or the opposite show that this doesn't happen with the metaseq codebase? Think this could really help in finding out if there is a bug.

**Comment by shijie-wu** [comment]
> @shijie-wu, could you by any chance reproduce the result using the metaseq code base - or the opposite show that this doesn't happen with the metaseq codebase? Think this could really help in finding out if there is a bug :-)

@patrickvonplaten I tried to load OPT-1.3B into metaseq but unfortunately it doesn't work ([error message](https://gist.github.com/shijie-wu/bad54e60b74f4d5448a63c79c3f1939e)). I would love to conduct the end-to-end regression test once I figure out what went wrong. Similar issues (https://github.com/facebookresearch/metaseq/issues/78 and https://github.com/facebookresearch/metaseq/issues/31) have been raised before regarding loading checkpoint into metaseq.

**Comment by punitkoura** [comment]
@patrickvonplaten @shijie-wu I added an end to end integration test between metaseq and HF in this PR https://github.com/facebookresearch/metaseq/pull/214

The test confirms that the logits and predicted words for -
1) Model parallel checkpoint
2) Metaseq singleton checkpoint
and 3) HF checkpoint

are the same. Let me know if this test contains what you were looking for. Thanks!

**Comment by patrickvonplaten** [comment]
Wow awesome thanks a lot @punitkoura !

**Comment by suchenzang** [comment]
Completed in https://github.com/facebookresearch/metaseq/pull/214 - thanks @punitkoura !!
